src/t.jl

```diff
@@ -28,18 +28,18 @@ export OneSampleTTest, TwoSampleTTest, EqualVarianceTTest,
 @compat abstract type TTest <: HypothesisTest end
 @compat abstract type TwoSampleTTest <: TTest end
 
-pvalue(x::TTest; tail=:both) = pvalue(TDist(x.df), x.t; tail=tail)
-
-default_tail(test::TTest) = :both
+pvalue(x::TTest; tail=x.tail) = pvalue(TDist(x.df), x.t; tail=tail)
+tail(x::TTest)  = x.tail    # defaults set by constructors
```

Comment:
Rather than defining this here, can you just use these definitions for all `HypothesisTest` types, and add the needed fields to all tests? If `tail` doesn't make sense for some types, you can then override the method (to a hardcoded value) instead of adding the field.